ddtrace/internal/writer.py

```diff
@@ -489,16 +492,20 @@ def write(self, spans=None):
 
     def flush_queue(self, raise_exc=False):
         # type: (bool) -> None
+        raise_exc = raise_exc or self._raise_exc
         try:
             try:
                 n_traces = len(self._encoder)
                 encoded = self._encoder.encode()
                 if encoded is None:
                     return
             except Exception:
-                log.error("failed to encode trace with encoder %r", self._encoder, exc_info=True)
-                self._metrics_dist("encoder.dropped.traces", n_traces)
-                return
+                if raise_exc:
+                    raise
```

Comment:
Note, if this raises an exception we'll raise an exception in the thread and will cause the thread to crash. We do not reset the service status so we'll never re-start the writer thread if there is an exception.

Reply:
I think we should omit this change for now.

Continue to catch and log the exception but do not raise since it can cause unintended side effects with the background writer thread (get it into a state where it is not working).

At the very least we have the changes to raise an exception on span finish if encoding fails which is the main place where that is likely to happen, and the exception is directly correlated to the event that caused it to fail (rather than randomly up to 1 sec into the future).